redhat-upgrade-tool.py

```diff
@@ -336,13 +336,21 @@ def handle_sigwinch(signum, frame):
     # Run the preuprade scripts if present
     if os.path.isdir(preupgrade_script_path):
         scripts = sorted(rlistdir(preupgrade_script_path))
+        failed = False
+        failed_scripts = {}
         for s in scripts:
             if os.access(s, os.X_OK):
                 try:
                     check_call(s)
                 except CalledProcessError as e:
-                    print("%s exited with status %d, exiting" % (s, e.returncode))
-                    sys.exit(1)
+                    failed_scripts[s] = e.returncode
+                    failed = True
+        if failed:
```

Comment:
You could use `if failed_scripts:` here instead of tracking the status in a second variable.